mavros/src/lib/uas_frame_conversions.cpp

```diff
@@ -18,26 +19,132 @@
 
 using namespace mavros;
 
-// Eigen based functions
+// Static quaternion needed for rotating between ENU and NED frames
+// +PI rotation around X (North) axis follwed by +PI/2 rotation about Z (Down)
+// gives the ENU frame.  Similarly, a +PI rotation about X (East) followed by 
+// a +PI/2 roation about Z (Up) gives the NED frame.  
+static const Eigen::Quaterniond NED_ENU_Q = UAS::quaternion_from_rpy(M_PI, 0.0, M_PI_2);
+
+// Static quaternion needed for rotating between aircraft and base_link frames
+// +PI rotation around X (Forward) axis transforms from Forward, Right, Down (aircraft) 
+// Fto Forward, Left, Up (base_link) frames.  
+static const Eigen::Quaterniond AIRCRAFT_BASELINK_Q = UAS::quaternion_from_rpy(M_PI, 0.0, 0.0);
 
-//! +PI rotation around X (Roll) axis give us ROS or FCU representation
-static const Eigen::Quaterniond FRAME_ROTATE_Q = UAS::quaternion_from_rpy(M_PI, 0.0, 0.0);
 
 //! Transform for vector3
-static const Eigen::Transform<double, 3, Eigen::Affine> FRAME_TRANSFORM_VECTOR3(FRAME_ROTATE_Q);
+//static const Eigen::Transform<double, 3, Eigen::Affine> FRAME_TRANSFORM_VECTOR3(FRAME_ROTATE_Q);
+
+
+Eigen::Quaterniond UAS::transform_orientation(const Eigen::Quaterniond &q, const UAS::STATIC_TRANSFORM transform)
+{
+	//Transform the attitude representation from frame to frame.  The proof for this transform can 
+	//be seen http://www.euclideanspace.com/maths/algebra/realNormedAlgebra/quaternions/transforms/
+		switch(transform){
+		case STATIC_TRANSFORM::NED_TO_ENU:
+		case STATIC_TRANSFORM::ENU_TO_NED:{
+			return NED_ENU_Q * q;
+			break;
+		}
+		case STATIC_TRANSFORM::AIRCRAFT_TO_BASELINK:
+		case STATIC_TRANSFORM::BASELINK_TO_AIRCRAFT:{
+			return q * AIRCRAFT_BASELINK_Q;
```

Comment:
Yes, the [link in the comment](http://www.euclideanspace.com/maths/algebra/realNormedAlgebra/quaternions/transforms/) sheds some insight as to why.

I've also verified with extensive testing.